mesonbuild/modules/hotdoc.py

```diff
@@ -415,7 +419,14 @@ def has_extensions(self, state, args, kwargs):
         # --c-include-directories
         KwargInfo(
             'dependencies',
-            ContainerTypeInfo(list, (Dependency, build.StaticLibrary, build.SharedLibrary)),
+            ContainerTypeInfo(list, (Dependency, build.StaticLibrary, build.SharedLibrary,
+                                     build.CustomTarget, build.CustomTargetIndex)),
+            listify=True,
+            default=[],
+        ),
+        KwargInfo(
+            'depends',
+            ContainerTypeInfo(list, (build.CustomTarget, build.CustomTargetIndex)),
```

Comment:
I think you should use `DEPENDS_KW` here for consistency. It accepts `BuildTarget` and `CustomTarget`, but not `CustomTargetIndex`... which is probably another bug actually...

Reply:
```
# TODO: CustomTargetIndex should be supported here as well
DEPENDS_KW: KwargInfo[T.List[T.Union[BuildTarget, CustomTarget]]] = KwargInfo(
```
Wondering why that wasn't added directly... @dcbaker ?

Reply:
Probably not the right time for changing this, just adding "since" is enough.